File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> traversal(TreeNode root,List<Integer> list){
        if(root==null){
            return list;
        }
        
        traversal(root.left,list);
        traversal(root.right,list);
        list.add(root.val);
        
        return list;
    }
    
    public List<Integer> postorderTraversal(TreeNode root) {
        return traversal(root,new ArrayList<Integer>());
    }
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorder(self, node, traversalList):
        if(node == None):
            return;
        
        self.postorder(node.left, traversalList);
        self.postorder(node.right, traversalList);
        traversalList.append(node.val);
        
        return;
    
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        traversalList = [];
        self.postorder(root, traversalList);
        return traversalList;
        